### PR TITLE
Fix Firebase AppCheck issue

### PR DESCRIPTION
**Pull request by ahmadrns**
Fix Firebase AppCheck issue, Firebase was not providing AppCheck token in debug mode when we use `AndroidProvider.playIntegrity` 

This PR resolve issue https://github.com/Pylons-tech/pylons/issues/1311

### CODE REVIEW

File: wallet/lib/main_prod.dart [hunk 1]
```diff
@@ -58,6 +57,23 @@ Future<void> main() async {
...
+Future<void> initializeAppCheck() async {
+  await FirebaseAppCheck.instance.activate(
+    webRecaptchaSiteKey: 'recaptcha-v3-site-key',
+  );
+  // FirebaseAppCheck when enforced would block incoming requests from Android emulator and iOS simulator too in debug mode.
+  // This kDebugMode check gets a debug token from FirebaseAppCheck which can then be added on the Firebase console
+  // So that the emulator and simulator can be allowed to access to Firebase AppCheck token.
+  if (kDebugMode) {
+    try {
+      const MethodChannel methodChannel = MethodChannel("method-channel");
```
Comment (kjawadDeveloper2) [inline wallet/lib/main_prod.dart:69]: The method channel must have a proper name.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

**Comment by kjawadDeveloper2** [comment]
@ahmadrns  what about ios?

**Comment by ahmadrns** [comment]
> @ahmadrns what about ios?

Added

**Comment by kjawadDeveloper2** [comment]
@ahmadrns  its also not working on the android physical device.

**Comment by faisalnaveedRNS** [comment]
@ahmadrns Please resolve issue,

**Comment by ahmadrns** [comment]
@kjawadDeveloper2 You will get something like this on your console when you build an application `Enter this debug secret into the allow list in the Firebase Console for your project: 26a77c38-5f8b-4328-abac-e60f15589f17`. just copy and paste this `debug secret` in firebase console and after that when you will make request to firebase appcheck token It will give you actual token.

**Comment by kjawadDeveloper2** [comment]
Works for android not for ios. I dont see anything in the logs for registering.

```
Launching lib/main_prod.dart on iPad Pro (12.9-inch) (5th generation) in debug mode...
lib/main_prod.dart:1
Xcode build done.                                           30.1s
remote notifications are not supported in the simulator
Connecting to VM Service at ws://127.0.0.1:58505/3KVNqjdRjTc=/ws
flutter: \^[[90m[🌎 Easy Localization] [DEBUG] Localization initialized<…>
flutter: \^[[90m[🌎 Easy Localization] [DEBUG] Start<…>
flutter: \^[[90m[🌎 Easy Localization] [DEBUG] Init state<…>
flutter: \^[[90m[🌎 Easy Localization] [DEBUG] Build<…>
flutter: \^[[90m[🌎 Easy Localization] [DEBUG] Init Localization Delegate<…>
flutter: \^[[90m[🌎 Easy Localization] [DEBUG] Init provider<…>
flutter: \^[[90m[🌎 Easy Localization] [DEBUG] Load Localization Delegate<…>
flutter: \^[[90m[🌎 Easy Localization] [DEBUG] Load asset from i18n<…>
```

**Review by kjawadDeveloper2 — APPROVED** [review]

**Review by Amean — APPROVED** [review]

**Pull request merged**